lib/DXIL/DxilMetadataHelper.cpp

```diff
@@ -1214,6 +1214,20 @@ Metadata *DxilMDHelper::EmitDxilFieldAnnotation(const DxilFieldAnnotation &FA) {
         m_pModule->GetDxilModule().GetOP()->GetResourcePropertiesType(),
         *m_pSM)));
   }
+  if (FA.HasBitFields()) {
+    const std::vector<DxilFieldAnnotation> &BitFields = FA.GetBitFields();
+    vector<Metadata *> MDBitFieldVals;
+    for (const DxilFieldAnnotation &BitField : BitFields) {
+      MDBitFieldVals.emplace_back(EmitDxilFieldAnnotation(BitField));
+    }
+    auto *BitFieldsMD = MDNode::get(m_Ctx, MDBitFieldVals);
+    MDVals.emplace_back(Uint32ToConstMD(kDxilFieldAnnotationBitFieldsTag));
+    MDVals.emplace_back(BitFieldsMD);
+  }
+  if (FA.HasBitFieldWidth()) {
+    MDVals.emplace_back(Uint32ToConstMD(kDxilFieldAnnotationBitFieldWidthTag));
+    MDVals.emplace_back(Uint32ToConstMD((unsigned)FA.GetBitFieldWidth()));
```

Comment:
why is the (unsigned) cast necessary? 
Doesn't GetBitFieldWidth() already return an unsigned type?

Reply:
Good catch. Removed.